libp2p/NodeTable.h

```diff
@@ -114,10 +114,20 @@ class NodeTable : UDPSocketEvents
     /// Returns distance based on xor metric two node ids. Used by NodeEntry and NodeTable.
     static int distance(NodeID const& _a, NodeID const& _b) { u256 d = sha3(_a) ^ sha3(_b); unsigned ret; for (ret = 0; d >>= 1; ++ret) {}; return ret; }
 
+    /// Open the udp socket and schedule execution of the discovery algorithm
+    void start();
+
     void stop()
     {
-        m_socket->disconnect();
-        m_timers.stop();
+        if (m_socket->isOpen())
+        {
+            // We "cancel" the timers by setting min_date_time rather than calling cancel() because
```

Comment:
Is it necessary to use this magic value for timer expiration? Would it be clearer if we instead just used a special flag like `m_stopped` to tell handlers that we're cancelling everything?

(Again I'm still not completely sure that this complication is necessary - why couldn't `Host` make sure that no handlers are being called (i.e. not `io_service::run()` is called) after `NodeTable` and its timers are destroyed. I beleive this is `Host`'s job)

Reply:
Yeah I think the flag would make things cleaner here - it would be similar to what `DeadlineOps::m_stopped` flag did before

Reply:
I actually like the magic value for timer expiration more because it's more self-contained (and I've commented its purpose so someone reading the code would know what it's used for) - we don't have to rely on an additional variable with potentially a different lifetime.